pennylane/grouping/optimize_measurements.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The main function for measurement reduction, `optimize_measurements` returns the partitions and
+corresponding necessary circuit post-rotations for a given list of Pauli words.
+"""
+
+from pennylane.grouping.group_observables import group_observables
+from pennylane.grouping.transformations import obtain_qwc_post_rotations_and_diagonalized_groupings
+
+
+def optimize_measurements(observables, coefficients=None, grouping="qwc", colouring_method="rlf"):
+    """Partitions then diagonalizes a list of Pauli words, facilitating simultaneous measurement of
+    all observables within a partition.
+
+    The input list of observables are partitioned into mutually qubit-wise commuting (QWC) or
+    mutually commuting partitions by approximately solving minimum clique cover on a graph where
+    each observable represents a vertice. The unitaries which diagonalize the partitions are then
+    found. See arXiv:1907.03358 and arXiv:1907.09386 for technical details of the QWC and fully
+    commuting measurement partitioning approaches respectively.
+
+    **Example usage:**
+
+    >>> observables = [qml.PauliY(0), qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(1)]
+    >>> coefficients = [1.43, 4.21, 0.97]
+    >>> post_rotations,diagonalized_groupings,grouped_coeffs = optimize_measurements(
+                                                                                     observables,
+                                                                                     coefficients,
+                                                                                     'qwc',
+                                                                                     'rlf'
+                                                                                     )
+    >>> print(post_rotations)
+    [[RY(-1.5707963267948966, wires=[0]), RY(-1.5707963267948966, wires=[1])],
+     [RX(1.5707963267948966, wires=[0])]]
```

Comment:
Yep, its currently consistent. The first element of `post_rotations` has 2 rotations since [X0X1] needs a rotation of the 0th and 1st qubits. The second grouping [Y0, Z1] only needs a rotation of the 0th qubit, so the second element of `post_rotations` has a single rotation.